martin/src/pg/pool.rs

```diff
@@ -96,3 +96,58 @@ async fn get_conn(pool: &Pool, id: &str) -> Result<Object> {
         .await
         .map_err(|e| PostgresPoolConnError(e, id.to_string()))
 }
+
+fn hide_pwd(conn_string: &str) -> String {
+    let mut new_str = conn_string.to_owned();
+
+    if let Some(at_idx) = conn_string.find('@') {
+        if let Some(colon_idx) = &conn_string[..at_idx].find("://") {
+            if let Some(pwd_idx) = &conn_string[colon_idx + 3..at_idx].find(':') {
+                let start = pwd_idx + colon_idx + 4;
+                let end = at_idx;
+                new_str.replace_range(start..end, &"*".repeat(end - start));
+            }
+        }
+    } else if let Some(pwd_idx) = conn_string.find("password=") {
```

Comment:
Seems there's still a bug here

Reply:
no no, this is not a good way to do this. Regex is built specifically for this type of jobs, or a dedicated postgres connection string parser (which is part of the postgress connection lib, so hard to reuse in a better way).  I did submit a better solution to this (IMO) - #907

In general for this type of issues, take a look at [documentation](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) -- I would use something like this, you can try it at https://rustexp.lpil.uk/

`^postgres(?:ql)?://[^:@/?]+:([^:@/?]+)@` -- the password is captured as the first capture group

Reply:
The reason I don't like Regex is it's easy to write hard to read later..

Reply:
it's not that bad, and it does save you in many cases, so highly advise to embrace it ... sparingly! ;)

Reply:
TBH, I don't think the code of this PR is easier to read / understand -- string parsing in code is notoriously non-trivial to understand either

Reply:
Agree. I like your #907 🤜

Reply:
> TBH, I don't think the code of this PR is easier to read / understand -- string parsing in code is notoriously non-trivial to understand either

TBH I'm exhausted to write this PR🤣